src/main/java/ch/njol/skript/expressions/ExprSentCommands.java

```diff
@@ -0,0 +1,158 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.events.EvtPlayerCommandSend;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerCommandSendEvent;
+import org.eclipse.jdt.annotation.Nullable;
+import org.skriptlang.skript.lang.structure.Structure;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+@Name("Sent Commands")
+@Description({
+	"The commands that will be sent to the player in a <a href='events.html#send_command_list'>send commands to player event</a>.",
+	"Modifications will affect what commands show up for the player to tab complete. They will not affect what commands the player can actually run.",
+	"Adding new commands to the list is illegal behavior and will be ignored."
+})
+@Examples({
+	"on send command list:",
+		"\tset command list to command list where [input does not contain \":\"]",
+		"\tremove \"help\" from command list"
+})
+@Since("INSERT VERSION")
+@Events("send command list")
+public class ExprSentCommands extends SimpleExpression<String> {
+
+	static {
+		Skript.registerExpression(ExprSentCommands.class, String.class, ExpressionType.SIMPLE, "[the] [sent] [server] command[s] list");
+	}
+
+	private EvtPlayerCommandSend parent;
+	private Kleenean delay;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(PlayerCommandSendEvent.class))
+			return false;
+		Structure structure = getParser().getCurrentStructure();
+		if (!(structure instanceof EvtPlayerCommandSend))
```

Comment:
Can this actually happen while current event is set to `PlayerCommandSendEvent` or is it just for safety if so maybe we should assert instead?

Reply:
As of now, I don't think it can happen, but it could if some other skript event also triggered off of PlayerCommandSendEvent I think.